Azure Automanage using an ARM template scenario (#640)
microsoft/azure_arc: docs/azure_arc_jumpstart/azure_arc_servers/day2/arc_automanage/_index.md

---
type: docs
title: "Azure Automanage"
linkTitle: "Azure Automanage"
weight: 9
description: >
---

## Enable Azure Automanage on an Azure Arc enabled server using an ARM template

> The following README will guide you on how to onboard an Azure Arc enabled server on to [Azure Automanage](https://docs.microsoft.com/en-us/azure/automanage/automanage-virtual-machines#prerequisites), so you can follow best practices in reliability, security and management for virtual machines using Azure services such as [Azure Update Management](https://docs.microsoft.com/en-us/azure/automation/update-management/overview) and [Azure Backup](https://docs.microsoft.com/en-us/azure/backup/backup-overview).

Reviewer: ```suggestion
The following README will guide you on how to onboard an Azure Arc enabled server on to [Azure Automanage](https://docs.microsoft.com/en-us/azure/automanage/automanage-virtual-machines#prerequisites), so you can follow best practices in reliability, security and management for Azure Arc enabled servers using Azure services such as [Azure Update Management](https://docs.microsoft.com/en-us/azure/automation/update-management/overview) and [Azure Backup](https://docs.microsoft.com/en-us/azure/backup/backup-overview).
```

?

> Azure Automanage removes the need to discover virtual machines manually and automatically onboards and configures certain services in Azure following best practices as defined in [Microsoft Cloud Adoption Framework for Azure](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/manage/best-practices). Azure services included in Auzure Automation are: [Azure Backup](https://docs.microsoft.com/en-us/azure/backup/backup-overview), [Azure Security Center](https://docs.microsoft.com/en-us/azure/security-center/security-center-introduction), [Azure Monitor](https://docs.microsoft.com/en-us/azure/azure-monitor/overview) and [Azure Automation](https://docs.microsoft.com/en-us/azure/automation/automation-intro).

Reviewer: ```suggestion
Azure Automanage removes the need to discover virtual machines manually and automatically onboards and configures certain services in Azure following best practices as defined in [Microsoft Cloud Adoption Framework for Azure](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/manage/best-practices). Azure services included in Azure Automation are: 
- [Azure Backup](https://docs.microsoft.com/en-us/azure/backup/backup-overview)
- [Azure Security Center](https://docs.microsoft.com/en-us/azure/security-center/security-center-introduction)
- [Azure Monitor](https://docs.microsoft.com/en-us/azure/azure-monitor/overview)
- [Azure Automation](https://docs.microsoft.com/en-us/azure/automation/automation-intro)
```

Easier to edit later as more services become available and have a cleaner look.

> In this guide, you will use an ARM template to enable and configure Azure Automation on your Azure Arc enabled servers.

Reviewer: I think it needs to have a more descriptive explanation. Something like "By the end of this guide..." which also incl. Automanage.

Author: changed

> > **Note: This guide assumes you already deployed VMs or servers that are running on-premises or other clouds and you have connected them to Azure Arc but If you haven't, this repository offers you a way to do so in an automated fashion:**

Reviewer: What about HCI?

Author: added to both the begining and end of the readme

    ![Screenshot of Azure Portal showing Azure Arc enabled server detail](./02.png)

* [Install or update Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest). Azure CLI should be running version 2.7** or later. Use ```az --version``` to check your current installed version.

* Create Azure service principal (SP)

    To connect a VM or bare-metal server to Azure Arc, Azure service principal assigned with the "Contributor" role is required. To create it, login to your Azure account run the below command (this can also be done in [Azure Cloud Shell](https://shell.azure.com/)).

    ```shell
    az login
    az ad sp create-for-rbac -n "<Unique SP Name>" --role contributor
    ```

    For example:

    ```shell
    az ad sp create-for-rbac -n "http://AzureArcServers" --role contributor
    ```

    Output should look like this:

    ```json
    {
    "appId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "displayName": "AzureArcServers",
    "name": "http://AzureArcServers",
    "password": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "tenant": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
    }
    ```

>   > **Note: It is optional but highly recommended to scope the SP to a specific [Azure subscription and resource group](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest). To create an Automanage Account used by the Automanage services, you need the Owner or Contributor permissions on your subscription along with User Access Administrator roles.**

Reviewer: ```suggestion
  > **Note: It is optional but highly recommended to scope the SP to a specific [Azure subscription](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest). To create an Automanage Account used by the Automanage services, you need the Owner or Contributor permissions on your subscription along with User Access Administrator roles.**
```

See #647

## Automation Flow

> For you to get familiar with the automation and deployment flow, below is an explanation.
> 
> * User is editing the shell script to provide environment variables (1-time edit). These variables are being used throughout the deployment.
> 
> * The script will edit the ARM templates parameters file that will be used for the deployment. There are three ARM templates that will:
> 
>   * Create an Azure Automanage Account.
>   * Grant premissions to the Azure Automanage Accout.
>   * Enable Azure Automanage on the Azure Arc enabled server.

Reviewer: ```suggestion
For you to get familiar with the automation and deployment flow, below is an explanation.

* User is editing the shell script to provide environment variables (1-time edit). These variables are being used throughout the deployment.

* The script will edit the ARM templates parameters file that will be used for the deployment. There are three ARM templates that will:

  * Create an Azure Automanage Account.
  * Grant permissions to the Azure Automanage Account.
  * Enable Azure Automanage on the Azure Arc enabled server.
```

## Enable Azure Automanage on an Azure Arc enabled server

> * In order to keep your local environment clean and untouched, we will use Azure Cloud Shell to run the `arc_automanage.sh` shell script against the Azure Arc enabled server.

Reviewer: See comment on file names

  ![Screenshot showing Azure Cloud Shell](./03.png)

> * To run the automation, navigate to the [deployment folder](https://github.com/microsoft/azure_arc/tree/main/azure_arc_servers_jumpstart/automanage/artifacts) and edit [the script's](https://github.com/microsoft/azure_arc/tree/main/azure_arc_servers_jumpstart/automanage/artifacts/arc_automanage.sh) environment variables, you have to provide:

Reviewer: ```suggestion
* To run the automation, navigate to the [deployment folder](https://github.com/microsoft/azure_arc/tree/main/azure_arc_servers_jumpstart/automanage/artifacts) and edit [the script's](https://github.com/microsoft/azure_arc/tree/main/azure_arc_servers_jumpstart/automanage/artifacts/arc_automanage.sh) environment variables:
```

>   * **automanageAccountName:** name of your Automanage Account identity, make sure it is unique.
>   * **location:** Azure Region where your Arc enabled server is registered to.
>   * **machineName:** Name of your Azure Arc enabled server as it is shown in the Azure Portal.
>   * **configurationProfileAssignment:** refers to the environment of your Azure Arc enabled server as Azure Automanage has different profiles. Values can be "Production" or "DevTest".

Reviewer: ```suggestion
  * _`automanageAccountName`_: name of your Automanage Account identity, make sure it is unique.
  * _`location`_: Azure Region where your Arc enabled server is registered to.
  * _`machineName`_: Name of your Azure Arc enabled server as it is shown in the Azure Portal.
  * _`configurationProfileAssignment`_: refers to the environment of your Azure Arc enabled server as Azure Automanage has different profiles. Values can be "Production" or "DevTest".
```

This is a new way we are highlighting variables/parameters in the index file to make it standout in the scenario web page. See [here](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_app_svc/aks/aks_app_service_arm_template/#deployment).

Author: noted

>  ![Screenshot environment variables](./04.png)

Reviewer: See command on parameter name _resourceGroup_

Author: modified to resource_group

> * Still from the deployment folder run the below command:

Reviewer: ```suggestion
* From the deployment folder run the below command:
```

  ```shell
    . ./arc_automanage.sh
  ```

>  ![Scripts output](./05.png)

Reviewer: Top Sub Id is not mask

Author: masked

  > **Note: For the script to work properly you must run this command from the deployment folder. The extra dot is due to the shell script has an *export* function and needs to have the vars exported in the same shell session as the rest of the commands.**

> * After the script has finished its run you will have Azure Automanage enabled. You should be able to see the Azure Arc enabled Server under 'Automanage – Azure machine best practices' with the Status set in 'Configured'.

Reviewer: I would add an additional screenshot on how to get to that view

Author: added

  ![Azure Arc enabled server in Azure Automanage](./06.png)

  > **Note: it may take upto 30 minutes for the script to finish its run**

## Clean up environment

> Complete the following steps to clean up your environment. To disable Azure Automanage you will use the Azure portal, go to the Automanage – Azure virtual machine best practices page that lists all of your auto-managed VMs. Select the checkbox next to the Azure Arc enabled Server you want to disable from Automanage, then click on the Disable automanagement button.

Reviewer: ```suggestion
Complete the following steps to clean up your environment. To disable Azure Automanage you will use the Azure portal, go to the Automanage – Azure virtual machine best practices page that lists all of your auto-managed VMs. Select the checkbox next to the Azure Arc enabled Server you want to disable from Automanage, then click on the Disable _automanagement_ button.
```
```suggestion
Complete the following steps to clean up your environment. To disable Azure Automanage you will use the Azure portal, go to the Automanage – Azure virtual machine best practices page that lists all of your auto-managed VMs. Select the checkbox next to the Azure Arc enabled Server you want to disable from Automanage, then click on the Disable automanagement button.
```
```suggestion
Complete the following steps to clean up your environment. To disable Azure Automanage you will use the Azure portal, go to the Automanage – Azure virtual machine best practices page that lists all of your auto-managed VMs. Select the checkbox next to the Azure Arc enabled Server you want to disable from Automanage, then click on the Disable _automanagement_ button.
```

Azure Automanage removes the need to discover virtual machines manually and automatically onboards and configures certain services in Azure following best practices as defined in [Microsoft Cloud Adoption Framework for Azure](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/manage/best-practices). Azure services included in Azure Automanage are:

- [Azure Backup](https://docs.microsoft.com/en-us/azure/backup/backup-overview)
- [Azure Security Center](https://docs.microsoft.com/en-us/azure/security-center/security-center-introduction)
- [Azure Monitor](https://docs.microsoft.com/en-us/azure/azure-monitor/overview)
- [Azure Automation](https://docs.microsoft.com/en-us/azure/automation/automation-intro)

By the end of this guide, you will have an Azure Arc enabled Server with Azure Automanage enabled and configured following Microsoft Cloud Adoption Framework best practices for Dev/Test or Production environments.

> > **Note: This guide assumes you already deployed VMs or servers that are running on-premises or other clouds and you have connected them to Azure Arc but If you haven't, this repository offers you a way to do so in an automated fashion:**

Reviewer 2: ```suggestion
> **Note: This guide assumes you already deployed VMs or servers that are running on-premises or in other clouds and you have connected them to Azure Arc. If you haven't, this repository offers you a way to do so in an automated fashion:**
```

* **[AWS Ubuntu EC2 instance](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_servers/aws/aws_terraform_ubuntu/)**
* **[AWS Amazon Linux 2 EC2 instance](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_servers/aws/aws_terraform_al2/)**
* **[Azure Ubuntu VM](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_servers/azure/azure_arm_template_linux/)**
* **[Azure Windows VM](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_servers/azure/azure_arm_template_win/)**
* **[VMware vSphere Ubuntu VM](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_servers/vmware/vmware_terraform_ubuntu/)**
* **[VMware vSphere Windows Server VM](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_servers/vmware/vmware_terraform_winsrv/)**
* **[Vagrant Ubuntu box](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_servers/vagrant/local_vagrant_ubuntu/)**
* **[Vagrant Windows box](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_servers/vagrant/local_vagrant_windows/)**

## Prerequisites

* Clone the Azure Arc Jumpstart repository

    ```shell
    git clone https://github.com/microsoft/azure_arc.git
    ```

> * As mentioned, this guide starts at the point where you already deployed and connected VMs or bare-metal servers to Azure Arc. For this scenario, as can be seen in the screenshots below, we will be using a Google Cloud Platform (GCP) instance that has been already connected to Azure Arc and is visible as a resource in Azure.

Reviewer 2: ```suggestion
* As mentioned, this guide starts at the point where you already deployed and connected VMs or bare-metal servers to Azure Arc. For this scenario we will be using a Google Cloud Platform (GCP) instance that has been already connected to Azure Arc and is visible as a resource in Azure.
```

    ```shell
    az login
    az ad sp create-for-rbac -n "<Unique SP Name>" --role contributor
    ```

    For example:

    ```shell
    az ad sp create-for-rbac -n "http://AzureArcServers" --role contributor
    ```

    Output should look like this:

    ```json
    {
    "appId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "displayName": "AzureArcServers",
    "name": "http://AzureArcServers",
    "password": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "tenant": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
    }
    ```

  > **Note: It is optional but highly recommended to scope the SP to a specific [Azure subscription](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest). To create an Automanage Account used by the Automanage services, you need the Owner or Contributor permissions on your subscription along with User Access Administrator roles.**

## Automation Flow

For you to get familiar with the automation and deployment flow, below is an explanation.

> * User is editing the shell script to provide environment variables (1-time edit). These variables are being used throughout the deployment.

Reviewer 2: This is a bit different than other scenarios in that user has to edit the main script file directly instead of editing a vars.sh file that exports the environment variables which are used by the main scripts. 

Any reason for deviating from the approach used in the other scenarios that require environment vars?
